cookbook-react-native/recipes/buttons/custom-hightlight-button/index.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useCallback, memo, useState, useMemo } from 'react';
+import {
+  TouchableHighlight,
+  Image,
+  TouchableHighlightProps,
+  ViewStyle,
+  ImageStyle,
+  TextStyle,
+  ActivityIndicator,
+  View,
+  GestureResponderEvent
+} from 'react-native';
+import CustomText from '@textsRecipes/custom-text';
+import { CustomTextVariants } from '@textsRecipes/custom-text/constants';
+
+import styles from './styles';
+import { VARIANTS, OPACITY, BUTTON_TYPES } from './constants';
+import { getUnderlayColor, getStyles, getFontProp, getActivityIndicatorColor, getButtonType, getButtonStatus } from './utils';
+import { getCustomStyles } from './utils/styleUtils';
+
+interface Props extends CustomTextVariants {
+  activeOpacity?: number;
+  big?: boolean;
+  blocked?: boolean;
+  borderless?: boolean;
+  disabled?: boolean;
+  icon?: number;
+  iconStyle?: ImageStyle;
+  link?: boolean;
+  loading?: boolean;
+  onPress: TouchableHighlightProps['onPress'];
+  secondary?: boolean;
+  style?: ViewStyle;
+  title?: string;
+  textStyle?: TextStyle;
+  small?: boolean;
+  medium?: boolean;
+  large?: boolean;
+}
+
+const CustomHighlightButton = ({
+  activeOpacity,
+  big,
+  blocked,
+  disabled,
+  icon,
+  iconStyle,
+  link,
+  loading,
+  onPress,
+  secondary,
+  style,
+  title,
+  textStyle,
+  ...props
+}: Props) => {
+  const [focused, setFocused] = useState(false);
+  const customStyles = useCallback(() => getCustomStyles(VARIANTS, props, styles), [props]);
+  const customTextStyles = useCallback(() => getCustomStyles(VARIANTS, props, styles, 'Content'), [props]);
+  const handlePressIn = useCallback(() => setFocused(true), []);
+  const handlePressOut = useCallback(
+    (event: GestureResponderEvent) => {
+      if (onPress) onPress(event);
+      setFocused(false);
+    },
+    [onPress]
+  );
+  const buttonType = getButtonType(secondary, link);
+  const buttonStatus = getButtonStatus(focused, disabled);
+  const { BUTTON_STYLE, TEXT_STYLE, ICON_STYLE } = useMemo(
+    () => getStyles(buttonType, buttonStatus),
+    [buttonType, buttonStatus]
+  );
+  const fontProp = useMemo(() => getFontProp(link, big), [link, big]);
+  return loading ? (
+    <View style={[styles.containerLoading, style]}>
+      <ActivityIndicator size="large" color={getActivityIndicatorColor(buttonType)} />
+    </View>
+  ) : (
+    <TouchableHighlight
+      onPressIn={handlePressIn}
+      onPressOut={handlePressOut}
+      style={[styles.container, big && styles.bigContainer, BUTTON_STYLE, customStyles(), style]}
+      underlayColor={getUnderlayColor(buttonType)}
+      activeOpacity={activeOpacity}
+      disabled={blocked || disabled}>
```

Comment:
Why blocked and disabled? What is the difference between them?

Reply:
`blocked` has the same behaviour as disabled without showing any feedback to the user... we used that behavior when we block some buttons to avoid double clicks and stuff like that... I can remove that if you want to